drizzlepac/haputils/analyze.py

```diff
@@ -287,12 +287,19 @@ def analyze_data(input_file_list, log_level=logutil.logging.NOTSET):
             no_proc_key = CHINKEY
             no_proc_value = chinject
 
-        # Filter name which starts with "G" for Grism, "PR" for Prism, or
-        # "BLOCK" for internal calibration of SBC
+        # Filter name which starts with "BLOCK" for internal calibration of SBC
         # The sfilter variable may be the concatenation of two filters (F160_CLEAR)
-        split_sfilter = sfilter.split('_')
+        #
+        # Grism/Prism images are also IMAGING=SPECTROSCOPIC, suppress the "no processing"
+        # indicators to allow the Grism/Prism images to be minimally processed for
+        # keyword updates.  This was done as a retrofit to allow Grism/Prism images
+        # to have the same WCS solution as the direct images in the visit (same detector).
+        split_sfilter = sfilter.upper().split('_')
         for item in split_sfilter:
-            if item.startswith(('G', 'PR', 'BLOCK')):
+            if item.startswith(('G', 'PR')):
```

Comment:
It is not clear to me what purpose this block of code for 'G', 'PR' options serves.  Any reason it can not simply be removed?

Reply:
This code is needed here in order to process the Grism/Prism images.  The "filter" check is done last and is not part of the extended if/elif block.  The Grism/Prism images are also SPECTROSCOPIC, so even if we do not exclude the images based upon filter, they would be excluded based upon imaging mode.  To process the Grism/Prism data, I make sure the "do not process" sentinel is cleared.